trurl: init at 0.3

christoph-heiss opened the pull request:
###### Description of changes

This is a new tool for "translating and transposing URLs", made by curl's Daniel Sternberg.
Intented as a companion tool to curl.

Introduction blog post: https://daniel.haxx.se/blog/2023/04/03/introducing-trurl/

06-04-2023: Updated 0.2 -> 0.3, which was released in the meantime ([changelog](https://github.com/curl/trurl/releases/tag/trurl-0.3)).

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ocfox commented on pkgs/tools/networking/trurl/default.nix:
```diff
@@ -0,0 +1,32 @@
+  checkTarget = "test";
+
+  meta = with lib; {
+    description = "A command line tool for URL parsing and manipulation.";
```
```suggestion
    description = "A command line tool for URL parsing and manipulation";
```
The description should not end with a period, see
https://nixos.org/manual/nixpkgs/stable/#var-meta-description

christoph-heiss commented:
Thanks for the review!
Fixed your suggestion, also updated the homepage link as it got an actual page now.

frobware commented on pkgs/tools/networking/trurl/default.nix:
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchFromGitHub, runCommand, curl, perl }:
```
Is `runCommand` required here? I tried it without and I'm able to build successfully.

christoph-heiss replied:
Yeah, it is not required, it's merely a leftover.
Thanks, fixed!

christoph-heiss commented on pkgs/tools/networking/trurl/default.nix:
```diff
@@ -0,0 +1,32 @@
+
+  enableParallelBuilding = true;
+
+  propagatedBuildInputs = [ curl ];
```
Thanks, fixed!

pstn commented:
[trurl 0.3](https://github.com/curl/trurl/releases/tag/trurl-0.3)  has been tagged.

christoph-heiss committed:
trurl: init at 0.3

This is a new tool for "translating and transposing URLs, made by curl's
Daniel Sternberg.

Signed-off-by: Christoph Heiss <christoph@c8h4.io>

christoph-heiss commented:
Updated to 0.3 and added the `changelog` meta attribute. No other changes.

marcusramberg reviewed (approved):
lgtm
